src/libostree/ostree-bootloader-grub2.c

```diff
@@ -299,8 +299,25 @@ _ostree_bootloader_grub2_write_config (OstreeBootloader      *bootloader,
   g_autofree char *bootversion_str = g_strdup_printf ("%u", (guint)bootversion);
   g_autoptr(GFile) config_path_efi_dir = NULL;
   g_autofree char *grub2_mkconfig_chroot = NULL;
+  gboolean use_grub2_mkconfig = FALSE;
+  const gchar *grub_exec = NULL;
+
+#ifdef OSTREE_GRUB2_MKCONFIG
+  use_grub2_mkconfig = TRUE;
+#endif
+  /* Autotests can set this envvar to select which code path to test, useful for OS installers as well */
+  grub_exec = g_getenv ("OSTREE_GRUB2_EXEC");
+  if (grub_exec)
+    {
+      if (g_str_has_suffix(grub_exec, "-mkconfig"))
+        use_grub2_mkconfig = TRUE;
```

Comment:
Curious about this conditional - wouldn't someone who wants `grub2-mkconfig` be fine just setting `OSTREE_GRUB2_EXEC=grub2-mkconfig` explicitly?  Do we need to magically interpret `OSTREE_GRUB2_EXEC=banana-mkconfig` as `grub2-mkconfig`?